consent/strategy_default.go

```diff
@@ -745,10 +745,42 @@ func (s *DefaultStrategy) issueLogoutVerifier(w http.ResponseWriter, r *http.Req
 	// logout flow.
 	redir := s.c.LogoutRedirectURL().String()
 
-	// The hint must be set if it's an RP-initiated logout flow.
-	hint := r.URL.Query().Get("id_token_hint")
-	state := r.URL.Query().Get("state")
-	requestedRedir := r.URL.Query().Get("post_logout_redirect_uri")
+	var hint string
+	var state string
+	var requestedRedir string
+
+	if r.Method == http.MethodGet {
+		// The hint must be set if it's an RP-initiated logout flow.
+		hint = r.URL.Query().Get("id_token_hint")
+		state = r.URL.Query().Get("state")
+		requestedRedir = r.URL.Query().Get("post_logout_redirect_uri")
+	} else if r.Method == http.MethodPost {
+		// The hint must be set if it's an RP-initiated logout flow.
+		err := r.ParseForm()
```

Comment:
ParseForm decodes both query (when GET) and body (when POST) so I think the distinction is not necessary! If you use r.PostForm it will only include vales from the body, if you use r.Form it includes both!